### PR TITLE
Better error handling check var in error

### PR DESCRIPTION
**Pull request by albertlast**
Check the var if isset before we use them.

### CODE REVIEW

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
`!empty($user_info['is_guest'])` would be more concise.

File: Sources/Errors.php [hunk 1]
```diff
@@ -490,7 +490,7 @@ function log_error_online($error, $sprintf = array())
 	if (SMF == 'SSI' || SMF == 'BACKGROUND')
 		return;
 
-	$session_id = $user_info['is_guest'] ? 'ip' . $user_info['ip'] : session_id();
+	$session_id = !empty($user_info['is_guest']) && $user_info['is_guest'] ? 'ip' . $user_info['ip'] : session_id();
```
Comment (Oldiesmann) [inline Sources/Errors.php]: The `&& $user_info['is_guest']` part is redundant here. If $user_info['is_guest'] is true, !empty will also be true.

Comment (albertlast) [reply Sources/Errors.php]: this only a catch when $user_info doesn't exists that we don't try to access  $user_info['is_guest'].
I could create error wher this happen that $user_info was not there.

Comment (Oldiesmann) [reply Sources/Errors.php]: See https://www.php.net/empty.

The `!empty()` call will return true if `$user_info['is_guest']` is true (or set to some other "non empty" value). The call will return false if `$user_info['is_guest']` is false (or some other "empty" value) or doesn't exist in the first place (including if `$user_info` isn't set). There's no need to check whether `$user_info` is true again after the `!empty()` call.

**Review by Sesquipedalian — APPROVED** [review]

**Pull request merged**